src/engine/renderer/tr_shader.cpp

```diff
@@ -4533,6 +4571,14 @@ static void CollapseStages()
 				stages[ normalStage ].active = false;
 			}
 
+			if ( heightStage != -1 )
```

Comment:
The commit message says the heightmap is a per-stage attribute, but this looks like it's in a stage by itself?

Reply:
The `dpMaterial` DarkPlaces compatibility layer is not yet ported to the pre-collapsed stage code and I'm using Xonotic assets to test this feature, hence the temporary requirement to collapse a standalone heightmap Stage. There is no keyword available to mapper to use the standalone heightmap stage, only the per-stage attribute in any way, so there is no chance a mapper use the standalone heightmap stage by mistake and we can drop it one day.

Reply:
I just added comments in code about this.